processing/src/main/java/io/druid/query/extraction/StringFormatExtractionFn.java

```diff
@@ -33,14 +33,26 @@
 public class StringFormatExtractionFn extends DimExtractionFn
 {
   private final String format;
+  private final boolean nullAsNull;
+  private final boolean nullAsEmpty;
 
   @JsonCreator
   public StringFormatExtractionFn(
-      @JsonProperty("format") String format
+      @JsonProperty("format") String format,
+      @JsonProperty("nullAsNull") boolean nullAsNull,
+      @JsonProperty("nullAsEmpty") boolean nullAsEmpty
   )
   {
     Preconditions.checkArgument(!Strings.isNullOrEmpty(format), "format string should not be empty");
+    Preconditions.checkArgument(!(nullAsNull && nullAsEmpty), "nullAsNull and nullAsEmpty cannot be both true");
     this.format = format;
+    this.nullAsNull = nullAsNull;
+    this.nullAsEmpty = nullAsEmpty;
```

Comment:
there are 3 behaviors (see the original issue #2706),

- treat `null` as `"null"`, `stringFormat("[%s]")` returns `"[null]"` (current)
- treat `null` as `null`, `stringFormat("[%s]")` returns `null` (nullAsNull)
- treat `null` as `""`, `stringFormat("[%s]")` returns `"[]"` (nullAsEmpty)